plugin.rb

```diff
@@ -60,6 +68,15 @@ def translate
         render_json_error e.message, status: 422
       end
     end
+
+    private
+
+    def user_group_allowed?
+      authorized_groups = SiteSetting.restrict_translation_by_group.split('|')
+      authorized = current_user.groups.any? {|group| authorized_groups.include? group.name }
+      authorized = true if authorized_groups.empty?
```

Comment:
Perhaps perform the `authorized_groups.empty?` check before looking at `current_user.groups`. In the majority of cases, I'd expect `authorized_groups.empty?` to be true, so you can return early and avoid looking at the groups of the `current_user`.